presto-main/src/main/java/io/prestosql/sql/planner/plan/LateralJoinNode.java

```diff
@@ -63,6 +85,7 @@
      */
     private final List<Symbol> correlation;
     private final Type type;
+    private final Expression filter;
```

Comment:
This should be called `criteria` (see my other comment in this PR).

Reply:
Can we maintain it as `Optional<Expression>` like we maintain in `JoinNode` ?

Reply:
> Can we maintain it as Optional<Expression> like we maintain in JoinNode ?

I would prefer the opposite. To remove `Optional` from `JoinNode`. To me it is nested optional-like structure, two levels where you need to check for emptiness.